app/updatemanager_test.go

```diff
@@ -363,3 +366,203 @@ func TestUpdateManager(t *testing.T) {
+				ID: "foobar",
+			},
+			expected: false,
+		},
+		"equal IDs and Prioritys": {
+			base: &UpdateControlMap{
+				ID:       "foo",
+				Priority: 1,
+			},
+			other: &UpdateControlMap{
+				ID:       "foo",
+				Priority: 1,
+			},
+			expected: true,
+		},
+		"equal IDs and unequal Prioritys": {
+			base: &UpdateControlMap{
+				ID:       "foo",
+				Priority: 1,
+			},
+			other: &UpdateControlMap{
+				ID:       "foo",
+				Priority: 2,
+			},
+			expected: false,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			assert.Equal(t, test.expected, test.base.Equal(test.other))
+		})
+	}
+}
+
+func TestMapExpired(t *testing.T) {
+	// Insert a map with a stamp
+	testMap := NewControlMap()
+	cm := &UpdateControlMap{
+		ID:       "foo",
+		Priority: 1,
+	}
+	testMap.Insert(cm.Stamp(1))
+	// Wait for the map to expire
+	time.Sleep(2 * time.Second)
+	_, expired := testMap.Get(cm.ID)
+	assert.True(t, cm.Equal(expired[0]))
+}
+
+// Test that inserting a map which exists in the expired pool, does replace the
+// expired one in the active pool
+func TestInsertMatchingMap(t *testing.T) {
+	// Insert an expired map
+	testMapPool := NewControlMap()
+	cm := &UpdateControlMap{
+		ID:       "foo",
+		Priority: 1,
+		States: map[string]UpdateControlMapState{
+			"ArtifactInstall": UpdateControlMapState{
+				Action: "continue",
+			},
+		},
+	}
+	testMapPool.Insert(cm.Stamp(1))
+	time.Sleep(2 * time.Second)
+	testMapPool.Get("foo") // Inspect the map in order to expire the map
+	require.Equal(t, 1, len(testMapPool.Expired["foo"]))
+	require.True(t, cm.Equal(testMapPool.Expired["foo"][0]))
+	// Insert a matching map
+	cmn := &UpdateControlMap{
+		ID:       "foo",
+		Priority: 1,
+		States: map[string]UpdateControlMapState{
+			"ArtifactRebootEnter": UpdateControlMapState{
+				Action: "ArtifactRebootEnter",
+			},
+		},
+	}
+	testMapPool.Insert(cmn.Stamp(2))
+	// Map hould exist in the active map
+	assert.Equal(t, 1, len(testMapPool.Active["foo"]))
+	assert.Contains(t, testMapPool.Active["foo"][0].States, "ArtifactRebootEnter", testMapPool.Active["foo"])
+	// But not in the inactive anylonger
+	assert.Equal(t, 0, len(testMapPool.Expired["foo"]))
+}
+
+func TestRemovalOfMaps(t *testing.T) {
+	// Not implemented
+	t.Fail()
+}
+
+func TestCleanExpiredMaps(t *testing.T) {
+	testMapPool := NewControlMap()
+	cm := &UpdateControlMap{
+		ID:       "foo",
+		Priority: 1,
+		States: map[string]UpdateControlMapState{
+			"ArtifactInstall": UpdateControlMapState{
+				Action: "continue",
+			},
+		},
+	}
+	cmn := &UpdateControlMap{
+		ID:       "foo",
+		Priority: 2,
+		States: map[string]UpdateControlMapState{
+			"ArtifactInstall": UpdateControlMapState{
+				Action: "continue",
+			},
+		},
+	}
+	testMapPool.Insert(cm.Stamp(0))
+	testMapPool.Insert(cmn.Stamp(0))
+	testMapPool.Clear()
+	assert.Equal(t, 0, len(testMapPool.Active["foo"]), testMapPool.Active["foo"])
+	assert.Equal(t, 0, len(testMapPool.Expired["foo"]))
```

Comment:
Maybe mix in one active map too, so that we know it doesn't just clear all maps?